LCM/dsc/engine/ConfigurationManager/LocalConfigurationManager.c

```diff
@@ -102,15 +163,24 @@ void TaskQueueLoop()
     
     while (shouldShutdown == 0)
     {
-	pthread_mutex_lock(&g_TaskQueueMutex);
+    pthread_mutex_lock(&g_TaskQueueMutex);
+    MI_Boolean taskStarted = MI_FALSE;
 	if (g_TaskHead != NULL)
 	{
 	    Thread t;
 	    PAL_Uint32 ret;
 
 	    currentTask = g_TaskHead;
 	    g_TaskHead = currentTask->next;
-	    pthread_mutex_unlock(&g_TaskQueueMutex);
+        pthread_mutex_unlock(&g_TaskQueueMutex);
+        
+        Context_Invoke_Basic *nodeArgs = (Context_Invoke_Basic *)currentTask->params;
+        if(nodeArgs != NULL && nodeArgs->methodName != NULL)
+        {
+            taskStarted = MI_TRUE;
+            Tcslcpy(g_CurrentRunningMethodName, nodeArgs->methodName, LCM_MAX_PATH);
+        }
+        DSC_EventWriteMessageStartingDscOperation(g_CurrentRunningMethodName);
```

Comment:
>g_CurrentRunningMethodName [](start = 50, length = 26)

this is not protected, is that ok?

Reply:
it should be protected. fixing it.

---
In reply to: [138495782](https://github.com/Microsoft/PowerShell-DSC-for-Linux/pull/323#discussion_r138495782) [](ancestors = 138495782)